kyuubi-server/src/main/scala/org/apache/kyuubi/engine/hive/HiveProcessBuilder.scala

```diff
@@ -113,7 +114,24 @@ class HiveProcessBuilder(
   override def shortName: String = "hive"
 }
 
-object HiveProcessBuilder {
+object HiveProcessBuilder extends Logging {
   final val HIVE_HADOOP_CLASSPATH_KEY = "HIVE_HADOOP_CLASSPATH"
   final val HIVE_ENGINE_NAME = "hive.engine.name"
+
+  def apply(
+      appUser: String,
+      conf: KyuubiConf,
+      engineRefId: String,
+      extraEngineLog: Option[OperationLog]): HiveProcessBuilder = {
+    val modeStr = conf.get(ENGINE_DEPLOY_MODE)
+    DeployMode.fromString(conf.get(ENGINE_DEPLOY_MODE)) match {
+      case LocalMode => new HiveProcessBuilder(appUser, conf, engineRefId, extraEngineLog)
+      case YarnMode =>
+        warn(s"Hive on YARN model is experimental, please use it carefully. " +
+          s"Set `${ENGINE_DEPLOY_MODE.key}` to `${LocalMode.toString}` " +
+          s"if you want to use it on production.")
```

Comment:
both are experimental ...
```suggestion
        warn(s"Hive on YARN model is experimental。")
```